java/vector/src/main/java/org/apache/arrow/vector/ipc/ArrowWriter.java

```diff
@@ -53,6 +53,8 @@ public abstract class ArrowWriter implements AutoCloseable {
 
   protected static final Logger LOGGER = LoggerFactory.getLogger(ArrowWriter.class);
 
+  protected static final int compressionLevel = 3;
```

Comment:
Constants should use UPPER_SNAKE_CASE.

Reply:
And this should be clearly named DEFAULT_COMPRESSION_LEVEL

Reply:
The default level is tricky here:
- LZ4 default level is 1 with range from 1 to 9
- ZSTD default level is 3 with range from 1 to 22 (also negative levels -1 to -5 are supported)

There is a [discussion](https://github.com/apache/arrow/issues/35287) on the C++ side to add support of `Codec::Options`. Would it be good to do similar things on the Java side so we can accommodate different default levels per codec?

Reply:
Hmm, we could leave that for a different PR?

Or we could use Optional<Integer> here and switch between constructors that way?

Reply:
+1 for Optional

Reply:
> The default level is tricky here:
> 
> * LZ4 default level is 1 with range from 1 to 9
> * ZSTD default level is 3 with range from 1 to 22 (also negative levels -1 to -5 are supported)
> 
> There is a [discussion](https://github.com/apache/arrow/issues/35287) on the C++ side to add support of `Codec::Options`. Would it be good to do similar things on the Java side so we can accommodate different default levels per codec?

Uhmm, our current compression version only support compression level for Zstd not for LZ4. I was reviewing and is close the same as [Spark Compression Level Zstd](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/io/CompressionCodec.scala).

Reply:
> +1 for Optional

Could you help me on this about where o how?

Reply:
We can use an `Optional<int> level` parameter and pass it to specific codecs via `createCodec`. `Lz4CompressionCodec` and `ZstdCompressionCodec` can respect the value when it is set or use specific default level if value is unset.

BTW, I just checked that `FramedLZ4CompressorOutputStream` does not have the concept of level. It requires user to fine-tune the parameters: https://commons.apache.org/proper/commons-compress/apidocs/org/apache/commons/compress/compressors/lz4/FramedLZ4CompressorOutputStream.Parameters.html

Reply:
Updated